libnavigation-base/api/current.txt

```diff
@@ -1126,7 +1142,7 @@ package com.mapbox.navigation.base.trip.notification {
     method public int getNotificationId();
     method public void onTripSessionStarted();
     method public void onTripSessionStopped();
-    method public void updateNotification(com.mapbox.navigation.base.trip.model.RouteProgress? routeProgress);
```

Comment:
Wondering if this method should in the public interface if we're decreasing the scope of the `TripNotificationState`.

If developers would like to visualize some other property from `RouteProgress`, they would have to push it into the custom notification instance anyway. Wondering if we can decrease the surface of this interface by making the `MapboxTripSession` call `updateNotification` only if the instance is `MapboxTripNotification`, and not for all implementations.

What do you think @cafesilencio?

Reply:
With the current implementation the TripNotification::updateNotification takes a nullable TripNotificationState.  Internally null is a valid value and the behavior changes depending on whether the value is null. Looking at MapboxTripSession::updateRouteProgress, your proposal would mean a nulled route progress would need to to map to a null value passed into the TripService::updateNotification. That doesn't seem like a natural thing to do, checking for a null and then passing in a null value knowing that null is valid argument.  When I see a method that takes a nullable argument I usually interpret that to mean the method can handle a null value not that null is a valid state that determines the business logic.  So I'm going to make some adjustments that instead of using a nullable TripNotificationState instead will use real objects to represent valid values and the factory will produce the appropriate state based on the route progress or null route progress argument. I think it reduces ambiguity and makes it overall more type safe.  I'll ping you for a review after it's pushed.

Reply:
The refactor to remove the ambiguity looks good to me but it wasn't the initial intent of my above comment.

I was wondering whether `MapboxTripSession` could do:
```
if (tripNotification is MapboxTripNotification) {
  tripNotification.updateNotification(TripNotificaitonState)
}
```

and then we could actually remove the `updateNotification` from the `TripNotification` interface contract. Let's not block this PR over that discussion though, we can continue separately.

Reply:
Ok, i'll merge this.